src/main/java/org/commcare/formplayer/db/callbacks/LockLoggingCallback.java

```diff
@@ -0,0 +1,72 @@
+package org.commcare.formplayer.db.callbacks;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.flywaydb.core.api.callback.Callback;
+import org.flywaydb.core.api.callback.Context;
+import org.flywaydb.core.api.callback.Event;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.datasource.SingleConnectionDataSource;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+@Component
+public class LockLoggingCallback implements Callback {
+
+    private final Log log = LogFactory.getLog(LockLoggingCallback.class);
+
+    // Copied from org.flywaydb.core.internal.database.postgresql.PostgreSQLAdvisoryLockTemplate
+    private static final long LOCK_MAGIC_NUM =
+            (0x46L << 40) // F
+                    + (0x6CL << 32) // l
+                    + (0x79L << 24) // y
+                    + (0x77 << 16) // w
+                    + (0x61 << 8) // a
+                    + 0x79; // y
+
+    @Override
+    public boolean supports(Event event, Context context) {
+        return event.equals(Event.BEFORE_MIGRATE);
+    }
+
+    @Override
+    public boolean canHandleInTransaction(Event event, Context context) {
+        return true;
+    }
+
+    private void log(Event event, String message, Object ...params) {
+        List<Object> allParams = new ArrayList<Object>(Arrays.asList(params));
+        allParams.add(0, event);
+        log.info(String.format("[%s] " + message, allParams.toArray()));
+    }
+
+    @Override
+    public void handle(Event event, Context context) {
+        String tableName = String.format("\"public\".\"%s\"", context.getConfiguration().getTable());
+        long lockName = LOCK_MAGIC_NUM + tableName.hashCode();
+        this.log(event, String.format("Lock name for this process: %s", lockName));
+
+        JdbcTemplate jdbcTemplate = new JdbcTemplate(new SingleConnectionDataSource(context.getConnection(), true));
```

Comment:
Do we need to wrap this with a finally of some sort? Want to make sure we don't leave a lingering connection around.

Reply:
This connection is managed by Flyway so there isn't anything we need to do to close it. This uses the same pattern as the migrations.

Reply:
great, thanks for confirming!